apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/plugins/editor-template-classes/index.js

```diff
@@ -24,15 +24,16 @@ const EditorTemplateClasses = withSelect( ( select ) => {
 	return { templateClasses };
 } )( ( { templateClasses } ) => {
 	const blockListInception = setInterval( () => {
-		const blockListParent = document.querySelector( '.block-editor-writing-flow' );
+		const blockListParent = document.querySelector(
+			'.editor-styles-wrapper > .block-editor-block-list__layout'
+		);
 
 		if ( ! blockListParent ) {
 			return;
 		}
 		clearInterval( blockListInception );
 
 		blockListParent.className = classNames(
-			'block-editor-writing-flow',
```

Comment:
Ah there is a small visual difference.  When we overwrite the className entirely I see extra margin around the header:

![Screen Shot 2021-08-03 at 10 32 50 AM](https://user-images.githubusercontent.com/28742426/128033959-c01a5b36-32d0-466e-96ea-9f72a26352f8.png)

However, when we replace (include blockListParent.className in the classNames function):

![Screen Shot 2021-08-03 at 10 34 35 AM](https://user-images.githubusercontent.com/28742426/128034150-3ee8d425-f6e7-44f8-b1d7-7283e06787f2.png)

Reply:
It does seem better to append classes rather than overwrite. @jeyip can you share your thinking about https://github.com/Automattic/wp-calypso/pull/55109/commits/d4bcf484982ffc4cfe8f3abc5478f76b501ac271?

Reply:
Oh I must've been confused yesterday. For some reason, I thought the change I made _was_ appending instead of overwriting, but yes, appending is ideal.

The reason I made the change was because if we include `blockListParent.className` as the first item, I'm seeing an odd bug where this is the result.

<img width="1451" alt="Screen Shot 2021-08-02 at 8 55 27 PM" src="https://user-images.githubusercontent.com/5414230/128053073-4b9a7fdb-063d-425b-8280-06bb3c0266f5.png">

Reply:
😮 that's no good! I'll see if I can track it down.

Also... note to future us: let's never inject things into the Gutenberg DOM like this again... it's very hard to maintain!

Reply:
> Also... note to future us: let's never inject things into the Gutenberg DOM like this again... it's very hard to maintain!

Totally agree 💯. I've been tinkering with the problem as well -- will post an update soon.

Reply:
@creativecoder I implemented a `useEffect` hook to ensure that the `setInterval` is only called once. Opinions?

```javascript
	useEffect( () => {
		const hasValidTemplateClass = templateClasses.some( ( templateClass ) =>
			Boolean( templateClass )
		);

		if ( hasValidTemplateClass ) {
			const blockListInception = setInterval( () => {
				const blockListParent = document.querySelector(
					'.editor-styles-wrapper > .block-editor-block-list__layout'
				);

				if ( ! blockListParent ) {
					return;
				}
				clearInterval( blockListInception );

				blockListParent.className = classNames(
					blockListParent.className,
					'a8c-template-editor fse-template-part',
					...templateClasses
				);
			} );
		}
	}, [ ...templateClasses ] );
```

Reply:
We could only add the class names if they aren't already present?

Im not sure why the function is being called duplicated times, but as a safety we could add a conditional early return before adding extra classNames:
```
if ( blockListParent.className.includes( 'a8c-template-editor fse-template-part' ) return;

blockListParent.className = classNames(
	blockListParent.className,
	'a8c-template-editor fse-template-part',
	...templateClasses
);
```
or something similar?

Reply:
> Also... note to future us: let's never inject things into the Gutenberg DOM like this again... it's very hard to maintain!

1000%.  Thats a large reason for it having been locked down on an outdated version, the maintenance hell.  I think this has been a lesson learned as a result of legacy FSE and some ETK mistakes, and reasons we are working to do things correctly now by aiming to extend from core itself, but its still important to emphasize for the future.